sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -257,11 +279,6 @@ public virtual Response<ConfigurationSetting> SetConfigurationSetting(Configurat
                 requestOptions.IfMatch = setting.ETag;
             }
 
-            return SetConfigurationSetting(setting, requestOptions, cancellationToken);
-        }
-
-        internal virtual async Task<Response<ConfigurationSetting>> SetConfigurationSettingAsync(ConfigurationSetting setting, MatchConditions requestOptions, CancellationToken cancellationToken = default)
-        {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope("Azure.Data.AppConfiguration.ConfigurationClient.SetConfigurationSetting");
```

Comment:
DiagnosticScope logic should move to the beginning of the method, just after parameter validations.

Reply:
Done